### ISSUE TITLE
* spec.validation.openAPIV3Schema.properties[spec].type: Required value: must not be empty for specified object fields; with Kubernetes 1.16

### ISSUE THREAD

**Issue by cyrilthank** [kubeflow/kubeflow#4184]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Executed the workaround in 
https://github.com/kubeflow/kubeflow/issues/3810#issuecomment-527687504

get errors like the below

no matches for kind "deployment" in version "extensions/v1beta1"
error: unable to recognize "STDIN": no matches for kind "Deployment" in version "apps/v1beta2"

**What did you expect to happen:**

all the pods to show up in
kubectl get pods -n kubeflow

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**
 Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard):
kfctl version: (use kfctl version):
kfctl v0.6.2-0-g47a0e4c7
Kubernetes platform: (e.g. minikube)
Kubernetes version: (use kubectl version):
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:54Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:50Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
OS (e.g. from /etc/os-release):
NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.3 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

**Comment by cyrilthank** [comment]
[kubeflow_error.txt](https://github.com/kubeflow/kubeflow/files/3650466/kubeflow_error.txt)

**Comment by cyrilthank** [comment]
not sure if this is related to
https://github.com/kubeflow/kubeflow/issues/2001#issuecomment-442587724
https://github.com/ksonnet/ksonnet/issues/895

Please help as this is now a show stopper for us

cyril@masternode:~/kf-poc$ kubectl get pods -n kubeflow
NAME                                          READY   STATUS             RESTARTS   AGE
admission-webhook-bootstrap-stateful-set-0    1/1     Running            0          16h
application-controller-stateful-set-0         1/1     Running            0          16h
jupyter-web-app-deployment-7b5f9c8b77-vqlwc   1/1     Running            0          16h
katib-controller-6dbf6db58d-xhzc8             1/1     Running            1          16h
profiles-deployment-7b79cd776c-4bs2r          2/2     Running            0          16h
seldon-operator-controller-manager-0          0/1     CrashLoopBackOff   189        15h
cyril@masternode:~/kf-poc$ 


cyril@masternode:~/kf-poc$ kubectl get nodes
NAME             STATUS   ROLES    AGE   VERSION
masternode.com   Ready    master   21h   v1.16.0
slavenode1       Ready    <none>   17h   v1.16.0
cyril@masternode:~/kf-poc$

**Comment by cyrilthank** [comment]
cyril@masternode:~/kf-poc$ kubectl get pods -n kubeflow
NAME                                          READY   STATUS             RESTARTS   AGE
admission-webhook-bootstrap-stateful-set-0    1/1     Running            0          16h
application-controller-stateful-set-0         1/1     Running            0          16h
jupyter-web-app-deployment-7b5f9c8b77-vqlwc   1/1     Running            0          16h
katib-controller-6dbf6db58d-xhzc8             1/1     Running            1          16h
profiles-deployment-7b79cd776c-4bs2r          2/2     Running            0          16h
seldon-operator-controller-manager-0          0/1     CrashLoopBackOff   189        15h
cyril@masternode:~/kf-poc$ 


cyril@masternode:~/kf-poc$ kubectl get nodes
NAME             STATUS   ROLES    AGE   VERSION
masternode.com   Ready    master   21h   v1.16.0
slavenode1       Ready    <none>   17h   v1.16.0
cyril@masternode:~/kf-poc$

**Comment by cyrilthank** [comment]
[kubeflow_error_trace_1.txt](https://github.com/kubeflow/kubeflow/files/3662014/kubeflow_error_trace_1.txt)

**Comment by cyrilthank** [comment]
kubeflow@masternode:/nfsroot/kf-poc$ kubectl get pods -n kubeflow
NAME                                          READY   STATUS              RESTARTS   AGE
admission-webhook-bootstrap-stateful-set-0    1/1     Running             0          3m49s
application-controller-stateful-set-0         1/1     Running             0          4m3s
jupyter-web-app-deployment-7b5f9c8b77-vvgsr   1/1     Running             0          3m41s
katib-controller-6dbf6db58d-8hsv7             1/1     Running             0          3m30s
profiles-deployment-7b79cd776c-gmtm8          0/2     ContainerCreating   0          2m23s
seldon-operator-controller-manager-0          0/1     CrashLoopBackOff    1          2m20s
kubeflow@masternode:/nfsroot/kf-poc$ kubectl get pods -n kube-system
NAME                                     READY   STATUS    RESTARTS   AGE
coredns-5644d7b6d9-s8zk8                 1/1     Running   2          21h
coredns-5644d7b6d9-x7wf9                 1/1     Running   2          21h
etcd-masternode.com                      1/1     Running   2          21h
kube-apiserver-masternode.com            1/1     Running   2          21h
kube-controller-manager-masternode.com   1/1     Running   2          21h
kube-flannel-ds-amd64-8584j              1/1     Running   3          21h
kube-flannel-ds-amd64-zmxvm              1/1     Running   2          21h
kube-proxy-msrjp                         1/1     Running   3          21h
kube-proxy-s7ql9                         1/1     Running   2          21h
kube-scheduler-masternode.com            1/1     Running   2          21h
tiller-deploy-77855d9dcf-znqjc           1/1     Running   0          9m50s
kubeflow@masternode:/nfsroot/kf-poc$

**Comment by cyrilthank** [comment]
kubeflow@masternode:/nfsroot/kf-poc$ kubectl get nodes
NAME                     STATUS   ROLES    AGE   VERSION
kubeflowslavenode1.com   Ready    <none>   21h   v1.16.0
masternode.com           Ready    master   21h   v1.16.0

**Comment by Jabbo16** [comment]
Moving here to not disrupt the helm issue, It seems you also need to add a missing "selector" field to the `spec` field to fix the `missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec` errors.

```
spec:
  selector:
    matchLabels:
      app: yourapp
      name: yourname
```
Replace app and name values with whatever kubeflow wants, should be something like this: https://github.com/helm/helm/issues/6374#issuecomment-533840097
or this:
https://github.com/helm/helm/issues/6374#issuecomment-533185074
Just check whatever deployments are failing and edit manually the files

**Comment by cyrilthank** [comment]
Thanks @Jabbo16  for your patient replies

Can you please advise where/how to obtain the yaml files referenced in 

https://github.com/helm/helm/issues/6374#issuecomment-533840097
https://github.com/helm/helm/issues/6374#issuecomment-533185074

**Comment by Jabbo16** [comment]
The yaml files there are related to installing tiller, that suffers the same api deprecation issues with k8s 1.16.0.
You need to check now your kubeflow deployment yaml files (I guess they are inside the `kustomize` folder and manually add the `selector` field in the similar way of how it was also needed to be added to the tiller deployment yaml to work with k8s.
Just use that fixed deployment file as an example to edit your kubeflow ones.

**Comment by cyrilthank** [comment]
Thanks a lot @Jabbo16  for your patience

sorry if my ignorance is showing 

i did the following steps



and still getting the below errors

**selector errors**  not clear on where (in which yaml file and where within the yaml file to put the selector clause)

service/katib-db unchanged
error: error validating "STDIN": error validating data: ValidationError(Deployment.spec): missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec; if you choose to ignore these errors, turn validation off with --validate=false
service/katib-manager unchanged
service/katib-manager-rest unchanged
error: error validating "STDIN": error validating data: ValidationError(Deployment.spec): missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec; if you choose to ignore these errors, turn validation off with --validate=false
customres

service/katib-ui unchanged
error: error validating "STDIN": error validating data: ValidationError(Deployment.spec): missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec; if you choose to ignore these errors, turn validation off with --validate=false
serviceaccount/metadata-ui unchanged
role.rbac.authorization.k8s.io/metadata-ui unchanged
service/katib-suggestion-random unchanged
error: error validating "STDIN": error validating data: ValidationError(Deployment.spec): missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec; if you choose to ignore these errors, turn validation off with --validate=false


 **Non- selector class errors** 


customresourcedefinition.apiextensions.k8s.io/zipkins.config.istio.io unchanged
namespace/istio-system unchanged
error: error validating "STDIN": error validating data: [ValidationError(Mutatin                                                                                                 gWebhookConfiguration.webhooks[0]): missing required field "sideEffects" in io.k                                                                                                 8s.api.admissionregistration.v1.MutatingWebhook, ValidationError(MutatingWebhook                                                                                                 Configuration.webhooks[0]): missing required field "admissionReviewVersions" in                                                                                                  io.k8s.api.admissionregistration.v1.MutatingWebhook]; if you choose to ignore th                                                                                                 ese errors, turn validation off with --validate=false

The CustomResourceDefinition "seldondeployments.machinelearning.seldon.io" is invalid:
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.properties[children].items: Required value: must be specified
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[configMapKeyRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[fieldRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[resourceFieldRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[secretKeyRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[resources].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.type: Required value: must not be empty at the root

**Comment by cyrilthank** [comment]
for the file

"./suggestion/base/suggestion-bayesianoptimization-service.yaml"

i did the below

apiVersion: v1
kind: Service
metadata:
  name: katib-suggestion-bayesianoptimization
  selector:
   matchLabels:
    app: katib
    component: suggestion-bayesianoptimization
spec:
  type: ClusterIP
  ports:
    - port: 6789
      protocol: TCP
      name: api
  selector:
   matchLabels:
    app: katib
    component: suggestion-bayesianoptimization

"./suggestion/base/suggestion-bayesianoptimization-deployment.yaml"

i did the below

apiVersion: apps/v1
kind: Deployment
metadata:
  name: katib-suggestion-bayesianoptimization
  selector:
   matchLabels:
    app: katib
    component: suggestion-bayesianoptimization
spec:
  replicas: 1
  template:
    metadata:
      name: katib-suggestion-bayesianoptimization
      selector:
       matchLabels:
        app: katib
        component: suggestion-bayesianoptimization
    spec:
      containers:
      - name: katib-suggestion-bayesianoptimization
        image: gcr.io/kubeflow-images-public/katib/v1alpha2/suggestion-bayesianoptimization:v0.1.2-alpha-289-g14dad8b
        imagePullPolicy: IfNotPresent
        ports:
        - name: api
          containerPort: 6789


and i get the message

service/katib-suggestion-random configured
error when applying patch:

to:
Resource: "/v1, Resource=services", GroupVersionKind: "/v1, Kind=Service"
Name: "katib-suggestion-bayesianoptimization", Namespace: "kubeflow"
Object: &{map["apiVersion":"v1" "kind":"Service" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"labels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"},\"type\":\"ClusterIP\"}}\n"] "creationTimestamp":"2019-09-27T09:26:14Z" "name":"katib-suggestion-bayesianoptimization" "namespace":"kubeflow" "resourceVersion":"55475" "selfLink":"/api/v1/namespaces/kubeflow/services/katib-suggestion-bayesianoptimization" "uid":"516ff052-59ad-43c1-a826-7d52020bcff8"] "spec":map["clusterIP":"10.104.43.45" "ports":[map["name":"api" "port":'\u1a85' "protocol":"TCP" "targetPort":'\u1a85']] "selector":map["app":"katib" "component":"suggestion-bayesianoptimization"] "sessionAffinity":"None" "type":"ClusterIP"] "status":map["loadBalancer":map[]]]}
for: "STDIN": error when creating patch with:
original:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","selector":{"labels":{"app":"katib","component":"suggestion-bayesianoptimization"}}},"spec":{"ports":[{"name":"api","port":6789,"protocol":"TCP"}],"selector":{"app":"katib","component":"suggestion-bayesianoptimization"},"type":"ClusterIP"}}

modified:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"matchLabels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"matchLabels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}},\"type\":\"ClusterIP\"}}\n"},"name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","selector":{"matchLabels":{"app":"katib","component":"suggestion-bayesianoptimization"}}},"spec":{"ports":[{"name":"api","port":6789,"protocol":"TCP"}],"selector":{"matchLabels":{"app":"katib","component":"suggestion-bayesianoptimization"}},"type":"ClusterIP"}}

current:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"labels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"},\"type\":\"ClusterIP\"}}\n"},"creationTimestamp":"2019-09-27T09:26:14Z","name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","resourceVersion":"55475","selfLink":"/api/v1/namespaces/kubeflow/services/katib-suggestion-bayesianoptimization","uid":"516ff052-59ad-43c1-a826-7d52020bcff8"},"spec":{"clusterIP":"10.104.43.45","ports":[{"name":"api","port":6789,"protocol":"TCP","targetPort":6789}],"selector":{"app":"katib","component":"suggestion-bayesianoptimization"},"sessionAffinity":"None","type":"ClusterIP"},"status":{"loadBalancer":{}}}

for: "STDIN": unable to find api field in struct ObjectMeta for the json field "selector"
Error from server (Invalid): error when creating "STDIN": Deployment.apps "katib-suggestion-bayesianoptimization" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]


Please help

**Comment by uniuuu** [comment]
Hi @cyrilthank 

Try to check this requirement: 

> spec.selector has to match labels of spec.template.metadata.labels

source: [https://stackoverflow.com/questions/53304461/error-selector-does-not-match-template-labels](url)

**Comment by cyrilthank** [comment]
Thank you @uniuuu can you please advise what i could do to unblock and move forward with the kubeflow poc?

**Comment by uniuuu** [comment]
@cyrilthank 

I can see from the first post:
` Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3",
`
Did you upgraded to 1.16.0 after what step.
Could you post a full script of how to reproduce problem. I might catch on then to get root cause.

**Comment by cyrilthank** [comment]
Thanks @uniuuu  for your patient support as i navigate the kubeflow to get the poc up to sell the idea to others.

Attached the scripts (including the wrong steps i mistakenly executed)
[Kubeflow_script_steps_executed.txt](https://github.com/kubeflow/kubeflow/files/3669241/Kubeflow_script_steps_executed.txt)

**Comment by uniuuu** [comment]
@cyrilthank 

Please re-post content of suggestion-bayesianoptimization-service.yaml with its indentation.

> for the file
> 
> "./suggestion/base/suggestion-bayesianoptimization-service.yaml"
> 
> i did the below
> 
> apiVersion: v1
> kind: Service
> metadata:
> name: katib-suggestion-bayesianoptimization
> selector:
> matchLabels:
> app: katib
> component: suggestion-bayesianoptimization
> spec:
> type: ClusterIP
> ports:
> - port: 6789
> protocol: TCP
> name: api
> selector:
> matchLabels:
> app: katib
> component: suggestion-bayesianoptimization
> 
> "./suggestion/base/suggestion-bayesianoptimization-deployment.yaml"
> 
> i did the below
> 
> apiVersion: apps/v1
> kind: Deployment
> metadata:
> name: katib-suggestion-bayesianoptimization
> selector:
> matchLabels:
> app: katib
> component: suggestion-bayesianoptimization
> spec:
> replicas: 1
> template:
> metadata:
> name: katib-suggestion-bayesianoptimization
> selector:
> matchLabels:
> app: katib
> component: suggestion-bayesianoptimization
> spec:
> containers:
> - name: katib-suggestion-bayesianoptimization
> image: gcr.io/kubeflow-images-public/katib/v1alpha2/suggestion-bayesianoptimization:v0.1.2-alpha-289-g14dad8b
> imagePullPolicy: IfNotPresent
> ports:
> - name: api
> containerPort: 6789
>

**Comment by cyrilthank** [comment]
[suggestion-bayesianoptimization-service-and -deployment.zip](https://github.com/kubeflow/kubeflow/files/3669514/suggestion-bayesianoptimization-service-and.-deployment.zip)

**Comment by cyrilthank** [comment]
Attached the files

**Comment by cyrilthank** [comment]
Also attached the kubernetes install/config steps performed

**Comment by cyrilthank** [comment]
[Kubernetes_script_steps_executed.txt](https://github.com/kubeflow/kubeflow/files/3669828/Kubernetes_script_steps_executed.txt)

**Comment by cyrilthank** [comment]
Much appreciate your advice on what i am doing wrong as our kubeflow poc is now stalled

**Comment by uniuuu** [comment]
> [suggestion-bayesianoptimization-service-and -deployment.zip](https://github.com/kubeflow/kubeflow/files/3669514/suggestion-bayesianoptimization-service-and.-deployment.zip)

Check with this `https://github.com/kubeflow/manifests/blob/master/katib-v1alpha1/suggestion/base/suggestion-bayesianoptimization-deployment.yaml`
Delete spec.template.metadata.selector
Delete metadata.selector
Leave only one selector and under spec.selector
I suppose you have to have also spec.template.labels defined and selector should match the label name.
Check if your on latest commit. This one `https://github.com/kubeflow/manifests/commit/0842eb2b97f433bd39f1dc1882cf833cefa6c68f` provides fix for compatibility issue.

**Comment by cyrilthank** [comment]
[suggestion-bayesianoptimization-deployment.zip](https://github.com/kubeflow/kubeflow/files/3670104/suggestion-bayesianoptimization-deployment.zip)

can you please confirm if this modified file is ok

requesting since i am getting the below message  

configmap/metrics-collector-template unchanged
2019/09/30 09:35:28 got file 'suggestion-bayesianoptimization-deployment.yaml', but '/nfsroot/kf-poc/kustomize/suggestion/base/suggestion-bayesianoptimization-deployment.yaml' must be a directory to be a root
Error: accumulating resources: recursed accumulation of path 'base': accumulating resources: accumulating resources from 'suggestion-bayesianoptimization-deployment.yaml': YAML file [suggestion-bayesianoptimization-deployment.yaml] encounters a format error.
error converting YAML to JSON: yaml: line 5: mapping values are not allowed in this context

error: no objects passed to apply
customresourcedefinition.apiextensions.k8s.io/notebooks.kubeflow.org configured


though https://codebeautify.org/yaml-validator returns it as valid yaml

**Comment by uniuuu** [comment]
Please pull latest commit from https://github.com/kubeflow/manifests

**Comment by cyrilthank** [comment]
can you please advise based on the below messages if https://raw.githubusercontent.com/kubeflow/manifests/0842eb2b97f433bd39f1dc1882cf833cefa6c68f/katib-v1alpha2/suggestion/base/suggestion-bayesianoptimization-deployment.yaml
 worked.

clusterrolebinding.rbac.authorization.k8s.io/metrics-collector unchanged
configmap/metrics-collector-template unchanged
warning: error calculating patch from openapi spec: unable to find api field "selector" in io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta
service/katib-suggestion-grid configured
service/katib-suggestion-hyperband configured
service/katib-suggestion-nasrl configured
service/katib-suggestion-random configured
deployment.apps/katib-suggestion-bayesianoptimization created
error when applying patch:

to:
Resource: "/v1, Resource=services", GroupVersionKind: "/v1, Kind=Service"
Name: "katib-suggestion-bayesianoptimization", Namespace: "kubeflow"
Object: &{map["apiVersion":"v1" "kind":"Service" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"labels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"},\"type\":\"ClusterIP\"}}\n"] "creationTimestamp":"2019-09-27T09:26:14Z" "name":"katib-suggestion-bayesianoptimization" "namespace":"kubeflow" "resourceVersion":"55475" "selfLink":"/api/v1/namespaces/kubeflow/services/katib-suggestion-bayesianoptimization" "uid":"516ff052-59ad-43c1-a826-7d52020bcff8"] "spec":map["clusterIP":"10.104.43.45" "ports":[map["name":"api" "port":'\u1a85' "protocol":"TCP" "targetPort":'\u1a85']] "selector":map["app":"katib" "component":"suggestion-bayesianoptimization"] "sessionAffinity":"None" "type":"ClusterIP"] "status":map["loadBalancer":map[]]]}
for: "STDIN": error when creating patch with:
original:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","selector":{"labels":{"app":"katib","component":"suggestion-bayesianoptimization"}}},"spec":{"ports":[{"name":"api","port":6789,"protocol":"TCP"}],"selector":{"app":"katib","component":"suggestion-bayesianoptimization"},"type":"ClusterIP"}}

modified:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"matchLabels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"matchLabels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}},\"type\":\"ClusterIP\"}}\n"},"name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","selector":{"matchLabels":{"app":"katib","component":"suggestion-bayesianoptimization"}}},"spec":{"ports":[{"name":"api","port":6789,"protocol":"TCP"}],"selector":{"matchLabels":{"app":"katib","component":"suggestion-bayesianoptimization"}},"type":"ClusterIP"}}

current:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"labels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"},\"type\":\"ClusterIP\"}}\n"},"creationTimestamp":"2019-09-27T09:26:14Z","name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","resourceVersion":"55475","selfLink":"/api/v1/namespaces/kubeflow/services/katib-suggestion-bayesianoptimization","uid":"516ff052-59ad-43c1-a826-7d52020bcff8"},"spec":{"clusterIP":"10.104.43.45","ports":[{"name":"api","port":6789,"protocol":"TCP","targetPort":6789}],"selector":{"app":"katib","component":"suggestion-bayesianoptimization"},"sessionAffinity":"None","type":"ClusterIP"},"status":{"loadBalancer":{}}}

for: "STDIN": unable to find api field in struct ObjectMeta for the json field "selector"
Error from server (Invalid): error when creating "STDIN": Deployment.apps "katib-suggestion-grid" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "katib-suggestion-hyperband" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "katib-suggestion-nasrl" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "katib-suggestion-random" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]
customresourcedefinition.apiextensions.k8s.io/notebooks.kubeflow.org configured

**Comment by uniuuu** [comment]
You can see the errors are from another deployments katib-suggestion-grid, katib-suggestion-hyperband, katib-suggestion-nasrl, katib-suggestion-random.

Now I see from https://www.kubeflow.org/docs/started/k8s/kfctl-k8s-istio/ you apply https://raw.githubusercontent.com/kubeflow/kubeflow/v0.6-branch/bootstrap/config/kfctl_k8s_istio.0.6.2.yaml  which uses repo manifests https://github.com/kubeflow/manifests
That's mean that there has to be a new release of kubeflow or so for k8s v1.16

If you need to have it done without waiting try to reinstall kubeflow from source code if you can.
You can also try to apply the rest of files from https://github.com/kubeflow/manifests/tree/master/katib-v1alpha2/suggestion/base but you can see from https://github.com/kubeflow/manifests/commit/0842eb2b97f433bd39f1dc1882cf833cefa6c68f it's not only ones which were updated.

**Comment by cyrilthank** [comment]
Thanks a lot @uniuuu  for your amazing patience with me.

Currently since i am in a rush to 'demonstrate' a running kubeflow instance

can you please share the optimal steps i can perform without having to wait for dev to release fix

**Comment by cyrilthank** [comment]
Hi @uniuuu 

can you please advise for the attached list of errors where i may find similar manifest files

requesting since i am hoping if we can get those manifest files i can get the centralboard working
currently the centralboard url localhost:31380  localhost:8080 is not coming up and i am unable to demonstrate a working instance of kubeflow to the stakeholders.

[errors_30Sep2019.txt](https://github.com/kubeflow/kubeflow/files/3670501/errors_30Sep2019.txt)

**Comment by uniuuu** [comment]
Sure, use this approach (didn't test it)
```
git clone https://github.com/kubeflow/manifests
cd manifests
git checkout 0842eb2b97f433bd39f1dc1882cf833cefa6c68f
find "file_name_from_error" .
```
Or in browser: 
https://ibb.co/WDwt3VhShowing ￼ 

But most probably it will stuck at some point because it's 

> 170 changed files with 647 additions and 295 deletions.

**Comment by cyrilthank** [comment]
kubeflow@masternode:/nfsroot/kf-poc$ kubectl get pod --all-namespaces|grep -vi Runn
NAMESPACE      NAME                                                          READY   STATUS              RESTARTS   AGE
default        your-garfish-prometheus-server-6cc989b798-82vls               1/2     CrashLoopBackOff    11         50m
istio-system   istio-policy-749ff546dd-vbq28                                 1/2     CrashLoopBackOff    14         50m
istio-system   istio-telemetry-6f6d8db656-8cpd7                              1/2     CrashLoopBackOff    16         50m
istio-system   prometheus-744f885d74-dkbtc                                   0/1     ContainerCreating   0          50m
kubeflow       metadata-db-5b7c98746-zhdg6                                   0/1     CrashLoopBackOff    11         50m
kubeflow       metadata-deployment-5dfbf84d69-hcml9                          0/1     CrashLoopBackOff    11         50m
kubeflow       metadata-deployment-5dfbf84d69-mffhq                          0/1     CrashLoopBackOff    11         50m
kubeflow       metadata-deployment-5dfbf84d69-sxlzq                          0/1     CrashLoopBackOff    11         50m
kubeflow       minio-d66d85f96-5fl2p                                         0/1     CrashLoopBackOff    11         50m
kubeflow       seldon-operator-controller-manager-0                          0/1     CrashLoopBackOff    11         45m


but still centralboard does not show

**Comment by cyrilthank** [comment]
Hi @uniuuu 

Following are the steps i executed 


  cd
  mkdir manifests_from_git
  cd manifests_from_git
  git clone https://github.com/kubeflow/manifests
  cd manifests
  git checkout 0842eb2b97f433bd39f1dc1882cf833cefa6c68f
 
   du -a |grep -i katib
    diff -bitw ./katib-v1alpha2/katib-db/base/katib-db-deployment.yaml /nfsroot/kf-poc/kustomize//katib-db/base/katib-db-deployment.yaml
   cp -prf  /nfsroot/kf-poc/kustomize//katib-db/base  /nfsroot/kf-poc/kustomize//katib-db/base.orig
    cp ./katib-v1alpha2/katib-db/base/katib-db-deployment.yaml /nfsroot/kf-poc/kustomize//katib-db/base/katib-db-deployment.yaml
   du -a |grep -i katib-manager
 
  du -a |grep -i katib-manager
 
  cp -ip ./katib-v1alpha2/katib-manager/base/katib-manager-deployment.yaml /nfsroot/kf-poc/kustomize/katib-manager/base
    cp -ip ./katib-v1alpha2/katib-manager/base/katib-manager-rest-deployment.yaml /nfsroot/kf-poc/kustomize/katib-manager/base
   du -a |grep -i katib-ui
   cp -ip ./katib-v1alpha2/katib-ui/base/katib-ui-deployment.yaml  /nfsroot/kf-poc/kustomize/katib-ui/base
    cd /nfsroot/kf-poc/
    ls -altr
   ./apply.sh


and following are the remaining errors

============error 1===================
destinationrule.networking.istio.io/istio-telemetry unchanged
Error from server (Invalid): error when creating "STDIN": Deployment.apps "grafana" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"grafana", "chart":"grafana", "heritage":"Tiller", "release":"istio"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "istio-citadel" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"security", "chart":"security", "heritage":"Tiller", "istio":"citadel", "release":"istio"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "istio-egressgateway" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"istio-egressgateway", "chart":"gateways", "heritage":"Tiller", "istio":"egressgateway", "release":"istio"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "istio-galley" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"galley", "chart":"galley", "heritage":"Tiller", "istio":"galley", "release":"istio"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "istio-ingressgateway" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"istio-ingressgateway", "chart":"gateways", "heritage":"Tiller", "istio":"ingressgateway", "release":"istio"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "istio-sidecar-injector" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"sidecarInjectorWebhook", "chart":"sidecarInjectorWebhook", "heritage":"Tiller", "istio":"sidecar-injector", "release":"istio"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "STDIN": Deployment.apps "istio-tracing" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"jaeger", "chart":"tracing", "heritage":"Tiller", "release":"istio"}: `selector` does not match template `labels`]
unable to recognize "STDIN": no matches for kind "PodDisruptionBudget" in version "policy/v1"

===================error 2 ====================

configmap/metrics-collector-template unchanged
warning: error calculating patch from openapi spec: unable to find api field "selector" in io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta


===============error 3

error: error when applying patch:

to:
Resource: "/v1, Resource=services", GroupVersionKind: "/v1, Kind=Service"
Name: "katib-suggestion-bayesianoptimization", Namespace: "kubeflow"
Object: &{map["apiVersion":"v1" "kind":"Service" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"labels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"},\"type\":\"ClusterIP\"}}\n"] "creationTimestamp":"2019-09-27T09:26:14Z" "name":"katib-suggestion-bayesianoptimization" "namespace":"kubeflow" "resourceVersion":"55475" "selfLink":"/api/v1/namespaces/kubeflow/services/katib-suggestion-bayesianoptimization" "uid":"516ff052-59ad-43c1-a826-7d52020bcff8"] "spec":map["clusterIP":"10.104.43.45" "ports":[map["name":"api" "port":'\u1a85' "protocol":"TCP" "targetPort":'\u1a85']] "selector":map["app":"katib" "component":"suggestion-bayesianoptimization"] "sessionAffinity":"None" "type":"ClusterIP"] "status":map["loadBalancer":map[]]]}
for: "STDIN": error when creating patch with:
original:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","selector":{"labels":{"app":"katib","component":"suggestion-bayesianoptimization"}}},"spec":{"ports":[{"name":"api","port":6789,"protocol":"TCP"}],"selector":{"app":"katib","component":"suggestion-bayesianoptimization"},"type":"ClusterIP"}}

modified:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"matchLabels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"matchLabels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}},\"type\":\"ClusterIP\"}}\n"},"name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","selector":{"matchLabels":{"app":"katib","component":"suggestion-bayesianoptimization"}}},"spec":{"ports":[{"name":"api","port":6789,"protocol":"TCP"}],"selector":{"matchLabels":{"app":"katib","component":"suggestion-bayesianoptimization"}},"type":"ClusterIP"}}

current:
{"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"katib-suggestion-bayesianoptimization\",\"namespace\":\"kubeflow\",\"selector\":{\"labels\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"}}},\"spec\":{\"ports\":[{\"name\":\"api\",\"port\":6789,\"protocol\":\"TCP\"}],\"selector\":{\"app\":\"katib\",\"component\":\"suggestion-bayesianoptimization\"},\"type\":\"ClusterIP\"}}\n"},"creationTimestamp":"2019-09-27T09:26:14Z","name":"katib-suggestion-bayesianoptimization","namespace":"kubeflow","resourceVersion":"55475","selfLink":"/api/v1/namespaces/kubeflow/services/katib-suggestion-bayesianoptimization","uid":"516ff052-59ad-43c1-a826-7d52020bcff8"},"spec":{"clusterIP":"10.104.43.45","ports":[{"name":"api","port":6789,"protocol":"TCP","targetPort":6789}],"selector":{"app":"katib","component":"suggestion-bayesianoptimization"},"sessionAffinity":"None","type":"ClusterIP"},"status":{"loadBalancer":{}}}

for: "STDIN": unable to find api field in struct ObjectMeta for the json field "selector"


=========error 4

The CustomResourceDefinition "seldondeployments.machinelearning.seldon.io" is invalid:
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.properties[children].items: Required value: must be specified
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[configMapKeyRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[fieldRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[resourceFieldRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[secretKeyRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[resources].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.type: Required value: must not be empty at the root
kubeflow@masternode:/nfsroot/kf-poc$

**Comment by cyrilthank** [comment]
[errors_30Sep2019_4errors_remaining.txt](https://github.com/kubeflow/kubeflow/files/3670712/errors_30Sep2019_4errors_remaining.txt)

can you please advise for the above 4 errors which yaml files need to be changed
requesting since i am unable to figure out the mapping between the error message and the yaml file for the above 4 errors

**Comment by cyrilthank** [comment]
kubeflow@masternode:/nfsroot/kf-poc$ kubectl get pod --all-namespaces|grep -vi Runnin
NAMESPACE      NAME                                                          READY   STATUS              RESTARTS   AGE
default        your-garfish-prometheus-server-6cc989b798-82vls               1/2     CrashLoopBackOff    16         81m
istio-system   istio-pilot-67f9bd57b-xzmph                                   0/2     CrashLoopBackOff    10         81m
istio-system   istio-policy-749ff546dd-vbq28                                 1/2     CrashLoopBackOff    26         81m
istio-system   istio-telemetry-6f6d8db656-8cpd7                              1/2     CrashLoopBackOff    26         81m
istio-system   prometheus-744f885d74-dkbtc                                   0/1     ContainerCreating   0          81m
kubeflow       katib-db-576c7f9f49-ftk7g                                     0/1     CrashLoopBackOff    7          11m
kubeflow       katib-manager-6b4f79fb56-jqnzg                                0/1     CrashLoopBackOff    7          11m
kubeflow       metadata-db-5b7c98746-zhdg6                                   0/1     CrashLoopBackOff    17         81m
kubeflow       metadata-deployment-5dfbf84d69-hcml9                          0/1     CrashLoopBackOff    17         81m
kubeflow       metadata-deployment-5dfbf84d69-mffhq                          0/1     CrashLoopBackOff    17         81m
kubeflow       metadata-deployment-5dfbf84d69-sxlzq                          0/1     CrashLoopBackOff    17         81m
kubeflow       minio-d66d85f96-5fl2p                                         0/1     CrashLoopBackOff    16         81m
kubeflow       ml-pipeline-64b79d75bd-jpxv2                                  0/1     Error               8          81m
kubeflow       ml-pipeline-persistenceagent-6bb55755bf-mv68m                 0/1     CrashLoopBackOff    12         81m
kubeflow       seldon-operator-controller-manager-0                          0/1     CrashLoopBackOff    17         76m
kubeflow@masternode:/nfsroot/kf-poc$

**Comment by cyrilthank** [comment]
in the git downloaded location i see seldondeployments.machinelearning.seldon.io-crd.yaml while in the installed location it is crd.yaml
can you please help

**Comment by uniuuu** [comment]
I hope kubeflow team will help with solution soon. 

> You can see the errors are from another deployments katib-suggestion-grid, katib-suggestion-hyperband, katib-suggestion-nasrl, katib-suggestion-random.
> 
> Now I see from https://www.kubeflow.org/docs/started/k8s/kfctl-k8s-istio/ you apply https://raw.githubusercontent.com/kubeflow/kubeflow/v0.6-branch/bootstrap/config/kfctl_k8s_istio.0.6.2.yaml which uses repo manifests https://github.com/kubeflow/manifests
> That's mean that there has to be a new release of kubeflow or so for k8s v1.16
> 
> If you need to have it done without waiting try to reinstall kubeflow from source code if you can.
> You can also try to apply the rest of files from https://github.com/kubeflow/manifests/tree/master/katib-v1alpha2/suggestion/base but you can see from [kubeflow/manifests@0842eb2](https://github.com/kubeflow/manifests/commit/0842eb2b97f433bd39f1dc1882cf833cefa6c68f) it's not only ones which were updated.

**Comment by cyrilthank** [comment]
Thanks @uniuuu 

is there a way you can "re-assign" this issue appropriately?

can you please advise if there is any other option we can try out?
more specifically can you please share the steps to install kubeflow from source code?
can we do this on top of existing kubernetes cluster?

**Comment by cyrilthank** [comment]
further confirmed that
 cp -ip ./seldon/seldon-core-operator/base/seldondeployments.machinelearning.seldon.io-crd.yaml /nfsroot/kf-poc/kustomize/seldon-core-operator/base
 mv -i crd.yaml crd.yaml.orig
mv seldondeployments.machinelearning.seldon.io-crd.yaml  crd.yaml

still yields the same message

The CustomResourceDefinition "seldondeployments.machinelearning.seldon.io" is invalid:
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[explainer].properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.properties[children].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[children].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[children].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].properties[endpoint].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[graph].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[configMapKeyRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[fieldRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[resourceFieldRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].properties[secretKeyRef].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.properties[valueFrom].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[env].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[svcOrchSpec].properties[resources].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.type: Required value: must not be empty for specified array items
* spec.validation.openAPIV3Schema.properties[spec].type: Required value: must not be empty for specified object fields
* spec.validation.openAPIV3Schema.type: Required value: must not be empty at the root
kubeflow@masternode:/nfsroot/kf-poc$

**Comment by cyrilthank** [comment]
@uniuuu @Jabbo16 can you please share some potential way forward out of this.

Requesting since our kubeflow poc is completely stalled at this point

**Comment by cyrilthank** [comment]
@uniuuu @Jabbo16 
can you please advise on alternative steps

requesting since based on @uniuuu 's below comments i am wondering if the kfctl_k8s_istio.0.6.2.yaml can be 'corrected' in some way and re-run

please help

Now I see from https://www.kubeflow.org/docs/started/k8s/kfctl-k8s-istio/ you apply https://raw.githubusercontent.com/kubeflow/kubeflow/v0.6-branch/bootstrap/config/kfctl_k8s_istio.0.6.2.yaml which uses repo manifests https://github.com/kubeflow/manifests
That's mean that there has to be a new release of kubeflow or so for k8s v1.16

**Comment by cyrilthank** [comment]
@uniuuu @Jabbo16
can you please advise if the below 
https://github.com/kubeflow/kubeflow/issues/4198#issuecomment-538048899

can be attempted on kubernetes 1.14

please advise as at present I am unable to go back to the team with kubeflow as a solution for us

**Comment by cyrilthank** [comment]
when we attempt the kubeflow 0.6.2 install on kubernetes 1.16.1

kindly advise

 
[kubernetes_upgrade_to_1_16_01_kubeflow_deploy_error_04Oct.txt](https://github.com/kubeflow/kubeflow/files/3689793/kubernetes_upgrade_to_1_16_01_kubeflow_deploy_error_04Oct.txt)

**Comment by cyrilthank** [comment]
Any updates on this please?

**Comment by cyrilthank** [comment]
> 
> 
> You can see the errors are from another deployments katib-suggestion-grid, katib-suggestion-hyperband, katib-suggestion-nasrl, katib-suggestion-random.
> 
> Now I see from https://www.kubeflow.org/docs/started/k8s/kfctl-k8s-istio/ you apply https://raw.githubusercontent.com/kubeflow/kubeflow/v0.6-branch/bootstrap/config/kfctl_k8s_istio.0.6.2.yaml which uses repo manifests https://github.com/kubeflow/manifests
> That's mean that there has to be a new release of kubeflow or so for k8s v1.16
> 
> If you need to have it done without waiting try to reinstall kubeflow from source code if you can.
> You can also try to apply the rest of files from https://github.com/kubeflow/manifests/tree/master/katib-v1alpha2/suggestion/base but you can see from [kubeflow/manifests@0842eb2](https://github.com/kubeflow/manifests/commit/0842eb2b97f433bd39f1dc1882cf833cefa6c68f) it's not only ones which were updated.

can you please advise which version kubernetes i could try this with and get the PoC moving

requesting since as per 
https://github.com/helm/helm/issues/6374#issuecomment-539978846
helm works fine with 1.15.4 
but for me kubeflow has been failing since 1.15.3 and 1.16.02

**Comment by jlewi** [comment]
@cyrilthank Kubeflow is currently tested on Kubernetes 1.14
https://github.com/kubeflow/manifests/blob/973fba5a2012e7c9e305092727b2da446a2b6b81/gcp/deployment_manager_configs/cluster-kubeflow.yaml#L36

It looks like you are hitting a bunch of validation errors with various Kubeflow manifests because of changes in behavior in Kubernetes 1.16.

Your best bet is to use Kubernetes 1.14 and to wait until Kubeflow as a whole is updated and tested against 1.16.

**Comment by jlewi** [comment]
Duplicate of #4228

**Comment by cyrilthank** [comment]
> @cyrilthank Kubeflow is currently tested on Kubernetes 1.14
> https://github.com/kubeflow/manifests/blob/973fba5a2012e7c9e305092727b2da446a2b6b81/gcp/deployment_manager_configs/cluster-kubeflow.yaml#L36
> 
> It looks like you are hitting a bunch of validation errors with various Kubeflow manifests because of changes in behavior in Kubernetes 1.16.
> 
> Your best bet is to use Kubernetes 1.14 and to wait until Kubeflow as a whole is updated and tested against 1.16.

Thanks a lot @jlewi for patiently chasing this down

 can you please share how i can reach 1.14 kubernetes

i mean should i download 1.16 install it and then downgrade or
whether there is a specific directive to mention version 1.14 during installation

**Comment by jlewi** [comment]
@cyrilthank If you need help installing Kubernetes please open up a new issue of kind question. We will do our best to help you but in general you should look upstream to your kubernetes provider for help with installing/upgrading/downloading Kubernetes.

If you are using one of the Cloud providers (GCP, AWS, Azure, OpenShift) that is working closely with Kubeflow then we can usually provide more help.

**Comment by cyrilthank** [comment]
Thank you @jlewi 
I am doing this on-prem
